rotkehlchen/chain/arbitrum_one/modules/umami/utils.py

```diff
@@ -0,0 +1,54 @@
+
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.arbitrum_one.modules.umami.constants import UMAMI_ASSET_VAULT_ABI
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.inquirer import Inquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def get_umami_vault_token_price(
+        inquirer: 'Inquirer',
+        vault_token: 'EvmToken',
+        evm_inquirer: 'EvmNodeInquirer',
+) -> Price | None:
+    """Gets token price for Umami vault.
+
+    Multiplies vault's price per share by the underlying token's USD price.
+
+    Args:
+        inquirer: For underlying token price lookup.
+        vault_token: Vault token to get price data for.
+        evm_inquirer: Inquirer instance for making contract calls.
+
+    Returns:
+        Price: USD price of underlying token, None if pricing fails
+    """
+    try:
+        price_per_share = evm_inquirer.call_contract(
+            contract_address=vault_token.evm_address,
+            abi=UMAMI_ASSET_VAULT_ABI,
+            method_name='pps',
+        )
+    except RemoteError as e:
+        log.error(f'Failed to get price per share for vault {vault_token.evm_address}: {e}')
+        return None
+
+    formatted_pps = token_normalized_value_decimals(
+        token_amount=price_per_share,
+        token_decimals=vault_token.decimals,
+    )
+    underlying_token = vault_token.underlying_tokens[0]
+    underlying_price = inquirer.find_usd_price(Asset(underlying_token.get_identifier(evm_inquirer.chain_id)))  # noqa: E501
```

Comment:
no need for a variable here

Also shouldn't you check that the vault has an udnerlying token in the first place and error log and return zero if it does not? Or is tht checked somewhere else?

```suggestion
    underlying_price = inquirer.find_usd_price(Asset(underlying_token[0].get_identifier(evm_inquirer.chain_id)))  # noqa: E501
```